manifests/init.pp

```diff
@@ -29,14 +29,22 @@
     info('puppet_agent performs no actions on Puppet 4+')
   }
   else {
-    class { '::puppet_agent::prepare': } ->
-    class { '::puppet_agent::install': } ->
-    class { '::puppet_agent::config': } ~>
-    class { '::puppet_agent::service': }
+    if $::architecture == 'x86' and $arch == 'x64' {
+      fail('Unable to install x64 on a x86 system')
+    }
+    if $::osfamily == 'windows' {
+      class { '::puppet_agent::osfamily::windows': }
```

Comment:
The current `prepare` class does reset some properties so they'll use new locations: `vardir`, `rundir`, `libdir`, `confdir`, `ssldir`, and moves the ssldir to the new default location.